### DISCUSSION TITLE
I'm trying to add a SSH key however "Key is invalid. You must supply a key in OpenSSH public key format" keeps appearing.

### DISCUSSION THREAD

**Discussion by linaaaahc** [community/community#28041] [New to GitHub]
I'm basically copying the following code into github from the linux terminal. Is there anything wrong with it?

ssh-ed25519 AAAAC3NzaC1lZDIlNTE5AAAAIKsxrHHSghWvO8MO+UlzmTHLxSp1BeuPX855Ywwdgi75 "my email"

It has 80 characters (except my email). How should I solve this? Thanks! :)

**Comment by airtower-luna** [comment]
The format looks correct, though you should know that the last part isn't necessarily a mail address, it's a comment (but the GitHub docs recommend using your email address). So my first guess is a problem with this part ([from the docs](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account#adding-a-new-ssh-key-to-your-account)):

> When copying your key, don't add any newlines or whitespace.

So make sure there are no spaces/newlines before or after your key.

**Reply by linaaaahc** [reply]
thanks! ive just checked that and for some reason it still doesn't work, but thanks for the suggestion :)

**Reply by gitstua** [reply]
There was a previous discussion which may help? https://github.com/orgs/community/discussions/23089

**Reply by linaaaahc** [reply]
yes, I did first look at that one before posting on here, but it seems like my problem doesn't correspond to any of the solutions offered there...Thanks :)

**Reply by supriyasingh711** [reply]
i am facing the same problem ? used ssh-rsa and ssh-ed also but none is working
in ssh-ed they are showing the error --Key is invalid. You must supply a key in OpenSSH public key format
and in ssh-ras they are showing the error--- the key is weak must have atleast 2048 bits whereas i used 4096 ..
I dont get whats the problem here ..please help me out have been trying this for two days??

**Reply by airtower-luna** [reply]
@supriyasingh711, that kinda sounds like you copied the wrong thing. You need to copy the *exact* content of the `.pub` file for the key you just generated. If you need more help, please describe the exact procedure you used to create the key so people can help.

**Reply by supriyasingh711** [reply]
no i have copied from id_ed25519.pub this folder only but still its showing not in OpenSSH format.

**Reply by gitstua** [reply]
Sorry to hear this has been an issue for several days for you.

There are a few commands shown in the documentation here which may be worth using to create a new key to test https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key

You can then follow the instructions to add to your GitHub account https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account#adding-a-new-ssh-key-to-your-account

**Reply by gitstua** [reply]
When I cat that key (not one I will use for production) I see this... 
<img width="548" alt="image" src="https://github.com/community/community/assets/25424433/ce558de3-e878-4252-ae92-b9d8cec8e9e8">

**Reply by airtower-luna** [reply]
There's no reason not to use that key for production. The screenshot shows only the *public* key (which is what you need to upload), and once you upload it [it'll be public anyway](https://docs.github.com/en/rest/users/keys?apiVersion=2022-11-28#list-public-keys-for-a-user). :wink:

**Comment by airtower-luna** [comment]
Odd guess, but…

> ssh-ed25519 AAAAC3NzaC1lZDIlNTE5AAAAIKsxrHHSghWvO8MO+UlzmTHLxSp1BeuPX855Ywwdgi75 "my email"

Are the quotes around the comment in the public key you're trying to upload? I don't see anything against them [in the OpenSSH format description](https://manpages.debian.org/bullseye/openssh-server/sshd.8.en.html#AUTHORIZED_KEYS_FILE_FORMAT), but they're certainly unusual, and I wonder if that'd give GitHub's parser trouble. :thinking:

**Reply by linaaaahc** [reply]
no they aren't! I have even tried uploading it without the comment and it still doesn't work...I'm honestly confused on what to do now lol I've tried everything...thank you!!

**Reply by airtower-luna** [reply]
I just tried to validate your key with [`ssh-keygen`](https://manpages.debian.org/bullseye/openssh-client/ssh-keygen.1.en.html). I copyied the public key as you posted it to `testkey.pub`, and then:

```
$ ssh-keygen -lf testkey.pub
testkey.pub is not a public key file.
```

So there is indeed *something* wrong. I can't even guess what, maybe a mistype while having the file open in an editor and now there's a missing/wrong letter? :sweat_smile: 

Anyway, you can also get the public key from the private key file (which you should of course keep private itself!):

```
ssh-keygen -yf ~/.ssh/id_ed25519
```

Adjust the filename if you stored it in a different location. That command will output your public key. I'd be curious what's different! :smile_cat:
